src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -157,57 +157,72 @@ private function parseDefinitions($content, $file)
         if (!is_array($content['services'])) {
             throw new InvalidArgumentException(sprintf('The "services" key should contain an array in %s. Check your YAML syntax.', $file));
         }
-        if (isset($content['services']['_defaults'])) {
-            if (!is_array($defaults = $content['services']['_defaults'])) {
-                throw new InvalidArgumentException(sprintf('Service defaults must be an array, "%s" given in "%s".', gettype($defaults), $file));
+        foreach ($content['services'] as $id => $service) {
+            $this->parseDefinition($id, $service, $file, $this->parseDefaults($content, $file));
+        }
+    }
+
+    /**
+     * Parses the "_defaults" key.
+     *
+     * @param array  $content
+     * @param string $file
+     *
+     * @return array
+     *
+     * @throws InvalidArgumentException
+     */
+    private function parseDefaults($content, $file)
+    {
+        if (!isset($content['services']['_defaults'])) {
+            return array();
+        }
+        if (!is_array($defaults = $content['services']['_defaults'])) {
+            throw new InvalidArgumentException(sprintf('Service defaults must be an array, "%s" given in "%s".', gettype($defaults), $file));
+        }
+        if (isset($defaults['alias']) || isset($defaults['class']) || isset($defaults['factory'])) {
+            @trigger_error('Giving a service the "_defaults" name is deprecated since Symfony 3.3 and will be forbidden in 4.0. Rename your service.', E_USER_DEPRECATED);
+
+            return array();
+        }
+
+        $defaultKeys = array('public', 'tags', 'inherit_tags', 'autowire');
+        unset($content['services']['_defaults']);
```

Comment:
btw, this tells me we are missing an assertion ensuring that this does not define a ``_default`` service